pkg/pki/x509/x509.go

```diff
@@ -213,10 +213,42 @@ func (k PublicKey) Subjects() []string {
 				names = append(names, strings.ToLower(name.String()))
 			}
 		}
+		otherName, _ := cryptoutils.UnmarshalOtherNameSAN(cert.Extensions)
+		if len(otherName) > 0 {
+			names = append(names, otherName)
+		}
 	}
 	return names
 }
 
+// Identities implements the pki.PublicKey interface
+func (k PublicKey) Identities() ([]string, error) {
+	// k contains either a key, a cert, or a list of certs
+	if k.key != nil {
+		pem, err := cryptoutils.MarshalPublicKeyToPEM(k.key)
+		if err != nil {
+			return nil, err
+		}
+		return []string{string(pem)}, nil
+	}
+
+	var cert *x509.Certificate
+	if k.cert != nil {
+		cert = k.cert.c
+	} else if len(k.certs) > 0 {
+		cert = k.certs[0]
+	}
+
+	var identities []string
+	pem, err := cryptoutils.MarshalPublicKeyToPEM(cert.PublicKey)
```

Comment:
same comment here: when people want a cert identity wouldn't they want to check the PEM encoded cert (not the extract pubkey)?

Reply:
Absolutely, I think what I was thinking when I implemented this was that we would extract relevant info from the certificate. but we might as well return the certificate along with parsed values, give the caller as much info as possible. 

The other option would be to return only the certificate, but I want to continue to extract useful values in case the caller only cares about a key value or SAN. Does this sound good?

Reply:
SG! For ephemeral certs it's weird to think of pinning either the key value or the cert value - but things might change if people start using fulcio more with their own long lived keys (?).